docs(examples): migrate to uv and PEP 723 inline script metadata

hasansezertasan opened the pull request:
Related to #704

## What

Migrate the runnable examples in `examples/` to [uv](https://docs.astral.sh/uv/) with [PEP 723](https://peps.python.org/pep-0723/) inline script metadata, so each example can be run directly with `uv run <example>.py` without a manual virtualenv or dependency install.

## Changes

- Add PEP 723 `# /// script` metadata blocks to the standalone examples (`barcodes.py`, `codepage_tables.py`, `qr_code.py`, `software_barcode.py`, `software_columns.py`, `weather.py`), declaring `requires-python`, dependencies, and a local editable `python-escpos` source.
- Tidy the example scripts (wrap top-level code in a `main()` entry point, minor cleanups) to fit the single-file script model.
- Replace the `examples/docker-flask/` example with `examples/flask-example/`, now managed by uv via `pyproject.toml` and `uv.lock`, with a uv-based `Dockerfile`.

## Notes

- No library/source changes — this only touches `examples/`.

hasansezertasan committed:
migrate(uv): migrate examples to use uv and PEP 723 Python Scripts

hasansezertasan committed:
Update Dockerfile and project files for UV migration

- Changed base image to use UV with Python 3.11.
- Updated working directory and file copying to align with UV requirements.
- Removed requirements.txt and replaced it with uv.lock and pyproject.toml for dependency management.
- Modified README to reflect new build and run commands.
- Set default command to run the application using UV.

hasansezertasan committed:
migrate(flask-example): create new example application with Flask and ESC/POS integration

- Added a new Flask application example in the `flask-example` directory.
- Implemented a simple printing functionality using `CupsPrinter`.
- Created a Dockerfile for building and running the application.
- Included `pyproject.toml` and `uv.lock` for dependency management.
- Updated README with build and run instructions.

belono commented on examples/barcodes.py:
```diff
@@ -1,11 +1,23 @@
+# /// script
+# requires-python = ">=3.12"
```
We should require the minimum version of Python we currently support, otherwise a user would not be able to run this example script on a version lower than 3.12 while we currently support Python >=3.9.
```suggestion
# requires-python = ">=3.9"
```

belono commented on examples/codepage_tables.py:
```diff
@@ -1,9 +1,12 @@
+# /// script
+# requires-python = ">=3.12"
```
Same on this file. We should require the minimum version of Python we currently support, otherwise a user would not be able to run this example script on a version lower than 3.12 while we currently support Python >=3.9.
```suggestion
# requires-python = ">=3.9"
```

belono commented on examples/qr_code.py:
```diff
@@ -1,21 +1,26 @@
+# /// script
+# requires-python = ">=3.12"
```
The same here. We should require the minimum version of Python we currently support, otherwise a user would not be able to run this example script on a version lower than 3.12 while we currently support Python >=3.9.
```suggestion
# requires-python = ">=3.9"
```

belono commented on examples/software_barcode.py:
```diff
@@ -1,9 +1,21 @@
+# /// script
+# requires-python = ">=3.12"
```
Here too. We should require the minimum version of Python we currently support, otherwise a user would not be able to run this example script on a version lower than 3.12 while we currently support Python >=3.9.
```suggestion
# requires-python = ">=3.9"
```

belono commented on examples/software_columns.py:
```diff
@@ -1,25 +1,38 @@
-""" Example for software_columns: Print text arranged into columns."""
+# /// script
+# requires-python = ">=3.12"
```
And here. We should require the minimum version of Python we currently support, otherwise a user would not be able to run this example script on a version lower than 3.12 while we currently support Python >=3.9.
```suggestion
# requires-python = ">=3.9"
```

belono commented on examples/weather.py:
```diff
@@ -1,4 +1,9 @@
-#!/usr/bin/python
+# /// script
+# requires-python = ">=3.12"
```
Same case. We should require the minimum version of Python we currently support, otherwise a user would not be able to run this example script on a version lower than 3.12 while we currently support Python >=3.9.
```suggestion
# requires-python = ">=3.9"
```

hasansezertasan committed:
Update examples/barcodes.py

Co-authored-by: Benito López <belono@users.noreply.github.com>

hasansezertasan committed:
Update examples/codepage_tables.py

Co-authored-by: Benito López <belono@users.noreply.github.com>

hasansezertasan committed:
Update examples/qr_code.py

Co-authored-by: Benito López <belono@users.noreply.github.com>

hasansezertasan committed:
Update examples/software_barcode.py

Co-authored-by: Benito López <belono@users.noreply.github.com>

hasansezertasan committed:
Update examples/software_columns.py

Co-authored-by: Benito López <belono@users.noreply.github.com>

hasansezertasan committed:
Update examples/weather.py

Co-authored-by: Benito López <belono@users.noreply.github.com>

patkan committed:
Merge remote-tracking branch 'origin/master' into pr/hasansezertasan/705

patkan committed:
format with black

hasansezertasan committed:
Merge remote-tracking branch 'origin/master' into migrate/uv/examples

Resolve modify/delete conflicts in examples/docker-flask/ by keeping the
deletion: the example was migrated to examples/flask-example/ using uv and
a pyproject.toml/uv.lock, superseding the old Dockerfile and requirements.txt.

hasansezertasan committed:
docs(examples): align flask-example deps with master versions

Bump Flask 2.3.2->3.1.3, Pillow 10.3.0->12.3.0, python-barcode 0.14.0->0.16.1,
qrcode 7.4.2->8.2.0, Werkzeug 3.0.6->3.1.6 to match master. Flask 3.1.3 pulls
in blinker>=1.9.0 and itsdangerous>=2.2.0, and Pillow 12.3.0 requires Python
>=3.10, so requires-python is raised accordingly. Regenerate uv.lock.

hasansezertasan committed:
fix(examples): pass bytes to _raw in codepage_tables

Adding type annotations made mypy check the function bodies, surfacing
latent str-to-_raw calls (_raw requires bytes, and Dummy.output joins with
b"", so str would fail at runtime). Encode the string arguments to bytes.

hasansezertasan committed:
ci: re-trigger CI (transient Codecov GPG failure)
